src/docgen/schema.ts

```diff
@@ -0,0 +1,335 @@
+   * Doc strings of the property.
+   */
+  readonly docs: string;
+
+  /**
+   * The type of the property.
+   */
+  readonly type: TypeSchema;
+
+  /**
+   * The default value of the property
+   */
+  readonly default?: string;
+}
+
+/**
+ * Describes a parameter.
+ */
+export interface ParameterSchema extends PropertySchema {}
+
+/**
+ * Common properties of a method.
+ */
+export interface MethodSchemaBase {
+
+  /**
+   * Unique id of the method.
+   */
+  readonly id: string;
+
+  /**
+   * Code snippet to display.
+   */
+  readonly snippet: string;
+
+  /**
+   * Method parameters.
+   */
+  readonly parameters: ParameterSchema[];
+}
+
+/**
+ * Describes a constructor.
+ */
+export interface InitializerSchema extends MethodSchemaBase {}
+
+/**
+ * Describes a method.
+ */
+export interface MethodSchema extends MethodSchemaBase {
+
+  /**
+   * Method name.
+   */
+  readonly name: string;
+}
+
+/**
+ * Describes a class.
+ */
+export interface ClassSchema {
+
+  /**
+   * Unique class id.
+   */
+  readonly id: string;
+
+  /**
+   * Class name.
+   */
+  readonly name: string;
+
+  /**
+   * Interfaces this class implements.
+   */
+  readonly interfaces: TypeSchema[];
+
+  /**
+   * Doc string for the class.
+   */
+  readonly docs: string;
+
+  /**
+   * Class initializer.
+   */
+  readonly initializer?: InitializerSchema;
+
+  /**
+   * Instance methods.
+   */
+  readonly instanceMethods: MethodSchema[];
+
+  /**
+   * Static functions.
+   */
+  readonly staticFunctions: MethodSchema[];
+
+  /**
+   * Properties.
+   */
+  readonly properties: PropertySchema[];
+
+  /**
+   * Constants.
+   */
+  readonly constants: PropertySchema[];
+}
+
+/**
+ * Describes a construct.
+ */
+export interface ConstructSchema extends ClassSchema {}
+
+/**
+ * Describes a struct.
+ */
+export interface StructSchema {
+
+  /**
+   * Unique struct id.
+   */
+  readonly id: string;
+
+  /**
+   * Struct name.
+   */
+  readonly name: string;
+
+  /**
+   * Doc string of the struct.
+   */
+  readonly docs?: string;
+
+  /**
+   * How to initialize the struct.
+   */
+  readonly initializer: string;
+
+  /**
+   * Properties.
+   */
+  readonly properties: PropertySchema[];
+}
+
+/**
+ * Describes an interface.
+ */
+export interface InterfaceSchema {
+
+  /**
+   * Unique interface id.
+   */
+  readonly id: string;
+
+  /**
+   * Interface name.
+   */
+  readonly name: string;
+
+  /**
+   * Interfaces that this interface extends.
+   */
+  readonly interfaces: TypeSchema[];
+
+  /**
+   * Types implementing this interface.
+   */
+  readonly implementations: TypeSchema[];
+
+  /**
+   * Doc string for this interface.
+   */
+  readonly docs: string;
+
+  /**
+   * Methods.
+   */
+  readonly instanceMethods: MethodSchema[];
+
+  /**
+   * Properties.
+   */
+  readonly properties: PropertySchema[];
+}
+
+/**
+ * Describes an enum member.
+ */
+export interface EnumMemberSchema {
+
+  /**
+   * Unique enum member id.
+   */
+  readonly id: string;
```

Comment:
Why not call this fqn?

Reply:
Thinking from the frontend app perspective, I was thinking `id` as more generic and explanatory than FQN. But I don't think that's actually the responsibility of this schema so `fqn` I think is more appropriate.